core/src/main/java/io/ultrabrew/metrics/data/ConcurrentIntTable.java

```diff
@@ -0,0 +1,457 @@
+// Copyright 2019, Oath Inc.
+// Licensed under the terms of the Apache License 2.0 license. See LICENSE file in Ultrabrew Metrics
+// for terms.
+
+package io.ultrabrew.metrics.data;
+
+import static io.ultrabrew.metrics.util.Commons.checkArgument;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sun.misc.Unsafe;
+
+public abstract class ConcurrentIntTable {
```

Comment:
This class duplicates a lot of code from `ConcurrentMonoidHashTable`, is there really nothing that can be shared between them (without sacrificing performance obviously)?